client/src/core/TableClass.js

```diff
@@ -218,6 +225,10 @@ export class Table extends mix().with(staticStoreMixin) {
         return h.nick || h.wcag;
       })
       .value();
+
+    this.get_dimensions();
+    this.get_group_by_func();
+    this.get_dimension_col_values_func();
```

Comment:
Not going to blame you or necessarily ask for this to be cleaned up right now, because you're just matching how the super legacy `TableClass` was written, just want to comment that this is not a good/modern way to deal with JS class construction. I believe this was likely mostly written pre-ES6, before the JS spec actually had real class support.

A lot of the properties set by these calls could be defined as keyword `get` getters. If they were actually expensive to compute, they could use memoization too. Otherwise, no point to do all this constructor-time calculation and indirect extension of the class. Just makes the whole file harder to read through.

Reply:
Used a getter for `dimensions` but leaving the `this.get_group_by_func()` and `this.get_dimension_col_values_func()` for now.